claims.go

```diff
@@ -0,0 +1,65 @@
+package jwtauth
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/pointerstructure"
+)
+
+// getClaim returns a claim value from allClaims given a provided claim string.
+// If this string is a valid JSONPointer, it will be interpreted as such to locate
+// the claim. Otherwise, the claim string will be used directly.
+func getClaim(logger log.Logger, allClaims map[string]interface{}, claim string) interface{} {
+	var val interface{}
+	var err error
+
+	if !strings.HasPrefix(claim, "/") {
+		val = allClaims[claim]
+	} else {
+		val, err = pointerstructure.Get(allClaims, claim)
+		if err != nil {
+			logger.Warn(fmt.Sprintf("unable to locate %s in claims: %s", claim, err.Error()))
+			return nil
+		}
+	}
+
+	// The claims unmarshalled by go-oidc don't use UseNumber, so there will
+	// be mismatches if they're coming in as float64 since Vault's config will
+	// be represented as json.Number. If the operator can coerce claims data to
+	// be in string form, there is no problem. Alternatively, we could try to
+	// intelligently convert float64 to json.Number, e.g.:
```

Comment:
go-oidc is actually quite small, it may be worth simply forking it so that we don't have to hack around this. If they change it upstream it changes the behavior so my guess is they won't want to do that, but we could ask. If we do fork we could put in a note essentially linking to the upstream and saying please don't rely on this library.

Reply:
I've created a task to track resolution of this. Since the handling works with strings, I don't think it need block the 1.1 beta. Handling numbers will be an improvement.